cmd/snap/cmd_interfaces.go

```diff
@@ -82,11 +90,24 @@ func (x *cmdInterfaces) Execute(args []string) error {
 	w := tabWriter()
 	defer w.Flush()
 	fmt.Fprintln(w, i18n.G("Slot\tPlug"))
+
+	wantedSnap := x.Positionals.Query.Snap
+
 	for _, slot := range ifaces.Slots {
-		if wanted := x.Positionals.Query.Snap; wanted != "" {
-			ok := wanted == slot.Snap
+		askedWithNick := false
+		if wantedSnap != "" {
+			ok := wantedSnap == slot.Snap
+			if !ok && wantedSnap == snap.Nickname(slot.Snap) {
+				askedWithNick = true
+				ok = true
+			}
+
 			for i := 0; i < len(slot.Connections) && !ok; i++ {
-				ok = wanted == slot.Connections[i].Snap
+				ok = wantedSnap == slot.Connections[i].Snap
+				if !ok && wantedSnap == snap.Nickname(slot.Connections[i].Snap) {
+					askedWithNick = true
+					ok = true
```

Comment:
I think the above checks could use a helper:
`matches, askedWithNick := isWantedSnap(snapname)`
and the helper would internaly check nickaname as well.